src/scripting/lua_widget_attributes.cpp

```diff
@@ -342,16 +428,316 @@ WIDGET_SETTER("marked_up_text", t_string, gui2::styled_widget)
+}
+
+WIDGET_GETTER("hint_text", t_string, gui2::combobox)
+{
+	return w.get_hint_text();
+}
+
+WIDGET_SETTER("hint_text", t_string, gui2::combobox)
+{
+	w.set_hint_text(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("hint_image", std::string, gui2::text_box)
+{
+	return w.get_hint_image();
+}
+
+WIDGET_SETTER("hint_image", std::string, gui2::text_box)
+{
+	w.set_hint_image(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("hint_text", t_string, gui2::text_box)
+{
+	return w.get_hint_text();
+}
+
+WIDGET_SETTER("hint_text", t_string, gui2::text_box)
+{
+	w.set_hint_text(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_SETTER("history", std::string, gui2::text_box)
+{
+	w.set_history(value);
+}
+
+WIDGET_GETTER("indentation_step_size", int, gui2::tree_view)
+{
+	return w.get_indentation_step_size();
+}
+
+WIDGET_SETTER("indentation_step_size", int, gui2::tree_view)
+{
+	if(value < 0) {
+		throw std::invalid_argument("indentation_step_size must be >= 0");
+	}
+	w.set_indentation_step_size(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::label)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::label)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::rich_label)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::rich_label)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::scroll_label)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::scroll_label)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("link_aware", bool, gui2::scroll_text)
+{
+	return w.get_link_aware();
+}
+
+WIDGET_SETTER("link_aware", bool, gui2::scroll_text)
+{
+	w.set_link_aware(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("link_color", std::string, gui2::label)
+{
+	return w.get_link_color().to_hex_string();
+}
+
+WIDGET_GETTER("link_color", std::string, gui2::rich_label)
+{
+	return w.get_link_color().to_hex_string();
+}
+
+WIDGET_GETTER("max_input_length", int, gui2::combobox)
+{
+	return w.get_max_input_length();
+}
+
+WIDGET_SETTER("max_input_length", int, gui2::combobox)
+{
+	if(value < 0) {
+		throw std::invalid_argument("max_input_length must be >= 0");
+	}
+	w.set_max_input_length(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("max_input_length", int, gui2::text_box)
+{
+	return w.get_max_input_length();
+}
+
+WIDGET_SETTER("max_input_length", int, gui2::text_box)
+{
+	if(value < 0) {
+		throw std::invalid_argument("max_input_length must be >= 0");
+	}
+	w.set_max_input_length(value);
+	INVALIDATE_LAYOUT();
+}
+
+WIDGET_GETTER("step_size", int, gui2::slider)
+{
+	return w.get_step_size();
+}
+
+WIDGET_SETTER("step_size", int, gui2::slider)
+{
+	if(value < 0) {
+		throw std::invalid_argument("step_size must be >= 0");
+	}
+	w.set_step_size(value);
+}
+
+WIDGET_GETTER("text_alignment", std::string, gui2::styled_widget)
```

Comment:
Sorry for the delayed reply, but it's possible to simplify this part by using these functions:
https://github.com/wesnoth/wesnoth/blob/7ca616f2ae8be849f84cc4b2e6fdc4479766c55b/src/gui/widgets/helper.cpp#L89-L118
Probably somebody can do it later, nothing big.

Reply:
Simplification done in 2e0f8c453e2caace51cd30804c6093727af94fda.